resources/helm/v2.4/istio-telemetry/prometheus/templates/_sd_namespace.tpl

```diff
@@ -0,0 +1,9 @@
+{{- define "sdNamespaces" }}
+namespaces:
+  names:
+{{- if gt (len .Values.prometheus.scrapingNamespaces) 0 }}
+{{ toYaml .Values.prometheus.scrapingNamespaces | indent 2 }}
+{{- else }}
+  - {{ .Values.global.istioNamespace }}
```

Comment:
Istio namespace must also be added to the `scrapingNamespaces` list in the prometheus reconciler, because currently it's being removed when SMMR is reconciled.

Steps to reproduce:
1. Create SMCP. At this point everything is fine and `namespaces.names` contains only `istio-system`.
2. Create SMMR with empty members. Prometheus' config map will be configured with `namespaces.names: []` and prometheus will log errors about watching resources at cluster scope level.
3. Update SMMR and set correct member namespaces. Now prometheus will no longer log errors, but `istio-system` won't be included in `namespaces.names`, so control plane namespace won't be monitored.

Reply:
The `istio-system` is now always included.